activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1231,10 +1231,17 @@ def check_constraint_options(table_name, expression, options) # :nodoc:
       #
       #   remove_check_constraint :products, name: "price_check"
       #
+      # Checks if the check constraint exists before trying to remove it. Will silently ignore check constraint that
+      # don't exist.
+      #
+      #   remove_check_constraint :products, name: "price_check", if_exists: true
+      #
       # The +expression+ parameter will be ignored if present. It can be helpful
       # to provide this in a migration's +change+ method so it can be reverted.
       # In that case, +expression+ will be used by #add_check_constraint.
       def remove_check_constraint(table_name, expression = nil, **options)
+        return if options[:if_exists] && !check_constraint_exists?(table_name, **options)
```

Comment:
This line should be after `return unless supports_check_constraints?` because this check should not be done if the database adapter does not support check constraint.

Reply:
@yahonda Updated.